psutil/arch/openbsd/proc.c

```diff
@@ -189,9 +189,11 @@ psutil_proc_cmdline(PyObject *self, PyObject *args) {
         Py_DECREF(py_arg);
     }
 
+    free(argv);
     return py_retlist;
 
 error:
+    free(argv);
```

Comment:
Shouldn't this be?

```c
if (argv != null)
     free(argv);
```

Reply:
`free()` handles NULL pointers, but I can add the `if (argv != NULL)` guard if you prefer that style.

https://en.cppreference.com/w/c/memory/free#:~:text=If%20ptr%20is%20a%20null%20pointer%2C%20the%20function%20does%20nothing

Reply:
Thank you. 
Side story: years ago psutil received a CVE due to a double `free()`, so I believe I started using this style since then.

Reply:
Double free is a separate issue: to avoid that you can use the coding style:
```
free(p);
p = NULL;
```